azext_iot/central/providers/device_template_provider.py

```diff
@@ -72,6 +74,17 @@ def list_device_templates(
             api_version=self._api_version,
         )
 
+        if compact:  # if asked for compact, just keep reduced info
```

Comment:
Wouldn't it be simpler to say compact means that only fields x, y, z are kept and create a separate template using that? That way you don't need to worry about having to pop unneeded fields if a new feature is added (assuming that the default compact fields stay the same).

Furthermore, this is weird that list can modify the template object here.

Reply:
followed you suggestions and refactored.